tools/cloudera-parcel-gcsconnector/create_parcel.sh

```diff
@@ -22,115 +22,248 @@
-else
-  echo "OS not in list, please provide valid OS name"
-fi
-
-## Display variable information input by user
-echo "Version number:${version}"
-echo "OS type:${OS}"
-echo "Filename:${filen}"
-mkdir -p ${filen^^}-$version/lib/hadoop/lib && mkdir -p ${filen^^}-$version/meta
-touch ${filen^^}-$version/meta/parcel.json
-
-##Download gcs connector jar and copy all folder content to parcel location
-wget https://storage.googleapis.com/hadoop-lib/gcs/gcs-connector-hadoop2-latest.jar
-cp * ${filen^^}-$version/lib/hadoop/lib/
-
-## Create parcel.json file required for parcel packaging
-cat >>${filen^^}-$version/meta/parcel.json<< EOL
+#######################################
+# Capture logs in create_parcel.log and exit
+# Globals:
+#   None
+# Arguments:
+#   None
+# Returns:
+#   None
+#######################################
+function graceful_exit() {
+  echo "-------------------------------">>${LOGDIR}/create_parcel.log
+  echo "From User: "$(whoami)>>${LOGDIR}/create_parcel.log
+  cat ${LOGDIR}/cparcel_output.log >> ${LOGDIR}/create_parcel.log
+  exit 1
+}
+
+#######################################
+# User Help function for irregular input parameters
+# Globals:
+#   None
+# Arguments:
+#   -f: Parcel name
+#   -v: Parcel version
+#   -o: Operating system
+#   -d: (Optional)Parcel deployment through Clodera Manager value : true
+# Returns:
+#   None
+#######################################
+function help {
+  echo 'Usage: create_parcel.sh -f <parcel_name> -v <version> -o <os_distro_suffix> [-d true]'
+  graceful_exit
+}
+
+# Main entry of the script
+main() {
+  BASEDIR=${HOME}/GCS-CDH-PARCEL
+  rm -rf ${BASEDIR}
+  mkdir -p ${BASEDIR}
+  LOGDIR=${HOME}/parcel-logs
+  mkdir -p ${LOGDIR}/
+  cd ${BASEDIR}
+  touch ${LOGDIR}/cparcel_error.log
+  touch ${LOGDIR}/cparcel_output.log
+
+  # Allowed OS parameter values and flag to check valid input
+  OS_FLAG=0
+  OS_VALUE_ARRAY=(el5 el6 el7 sles11 sles12 lucid precise trusty squeeze wheezy)
+
+  # Link to GCS connector jar file and flag to validate the file existence
+  GCSJAR_LINK="https://storage.googleapis.com/hadoop-lib/gcs/gcs-connector-hadoop2-latest.jar"
+  GCSJAR_FLAG="1"
+
+  NUM_ARG=$#
+  exec > >(tee -i ${LOGDIR}/cparcel_output.log)
+  exec 2>&1
+  echo $(date)
+
+  # *************
+  # Validating The PARAMETERS
+  # Regex variable for Parcel and version validity
+  PARCEL_FLAG='^[A-Za-z0-9]+$'
+  VERSION_FLAG='^[0-9]\.[0-9]\.[0-9]$'
+
+  # Getting the parameter values from command line
+  while getopts 'f:v:o:d:' OPTION; do
+    case "${OPTION}"
+    in
+      f) FILE_NAME=${OPTARG} ;;
+      v) VERSION=${OPTARG} ;;
+      o) OSTYPE=${OPTARG} ;;
+      d) PLACE_FILE=${OPTARG} ;;
+      :) help ;;
+      \?) help ;;
+      *) help ;;
+    esac
+  done
+  shift "$(($OPTIND -1))"
+
+  if [[ NUM_ARG -lt 6 ]] || [[ NUM_ARG -gt 8 ]]; then
+    echo 'Error: number of parameters is not correct.'
+    help
+  fi
+  if [[ ${FILE_NAME} =~ ${PARCEL_FLAG} && ${VERSION} =~ ${VERSION_FLAG} ]]; then
```

Comment:
Should this be an elif? Otherwise add a newline between if-blocks for readbility